public/app/plugins/datasource/loki/components/LokiExploreQueryEditor.tsx

```diff
@@ -0,0 +1,96 @@
+// Libraries
+import React, { memo, useState, useEffect } from 'react';
+import _ from 'lodash';
+
+// Types
+import { AbsoluteTimeRange, ExploreQueryFieldProps, ExploreMode } from '@grafana/data';
+import { LokiDatasource } from '../datasource';
+import { LokiQuery, LokiOptions } from '../types';
+import { LokiQueryField } from './LokiQueryField';
+import { useLokiSyntax } from './useLokiSyntax';
+import { LokiExploreExtraField } from './LokiExploreExtraField';
+
+type Props = ExploreQueryFieldProps<LokiDatasource, LokiQuery, LokiOptions>;
+
+export const LokiQueryEditor = memo(function LokiQueryEditor(props: Props) {
+  const { query, data, datasource, exploreMode, history, onChange, onRunQuery } = props;
+
+  let absolute: AbsoluteTimeRange;
+  if (data && !_.isEmpty(data.request)) {
+    const { range } = data.request;
+
+    absolute = {
+      from: range.from.valueOf(),
+      to: range.to.valueOf(),
+    };
+  } else {
+    absolute = {
+      from: Date.now() - 10000,
+      to: Date.now(),
+    };
+  }
+
+  const [maxLines, setMaxLines] = useState(query?.maxLines?.toString() || datasource?.maxLines?.toString());
+
+  const { isSyntaxReady, setActiveOption, refreshLabels, ...syntaxProps } = useLokiSyntax(
+    datasource.languageProvider,
+    absolute
+  );
+
+  function onChangeQueryLimit(value: string, override?: boolean) {
+    const { query, onChange, onRunQuery } = props;
+    if (onChange) {
+      const nextQuery = { ...query, maxLines: +value };
```

Comment:
If value is empty string this will set `maxLines: 0`. Is that a correct behaviour here?

Reply:
Changed it a little - now it falls back to `auto` aka limit of the datasource.